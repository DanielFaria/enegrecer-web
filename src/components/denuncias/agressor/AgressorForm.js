import React from 'react';
import Nome from '../../comum/nome';
import CampoTexto from '../../comum/campoTexto';
import Genero from '../../comum/genero';
import Data from '../../comum/data';
import * as ConstantesCSS from '../ConstantesCss';
import Raca from '../../comum/comboboxRaca';

const AgressorForm = () => (
  <div>
    <h3>Você se sente confortável em dividir conosco informações sobre o agressor?</h3>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      Nunc nisl mi, egestas ac ipsum sit amet,
      blandit posuere metus. Donec imperdiet ut mi sed posuere.
      Pellentesque et varius sapien.
    </p>

    <div className="row">
      <Nome id={'nome-agressor'} />
      <CampoTexto
        id={'cidade-agressor'}
        label={'Cidade'}
        maxLen={40}
        placeholder={''}
        type={'text'}
        divClasse={'input-field col s6'}
      />
    </div>

    <div className="row">
      <Genero
        id={'genero-agressor'}
        divClasse={`${ConstantesCSS.CLASSES_DIV_INPUT} col s6`}
      />

      <Data
        id={'data-nascimento-agressor'}
        label={'Data de Nascimento'}
        divClasse={`${ConstantesCSS.CLASSES_DIV_INPUT} col s6`}
      />
    </div>

    <div className="row">
      <Raca id={'raca-agressor'} somenteRacasVitima={false} />
    </div>

    <div className="row">
      <CampoTexto
        id={'descricao-agressor'}
        label={'Descrição do agressor'}
        maxLen={100}
        placeholder={''}
        type={'text'}
        divClasse={'input-field col s12'}
      />
    </div>
  </div>
);

export default AgressorForm;
