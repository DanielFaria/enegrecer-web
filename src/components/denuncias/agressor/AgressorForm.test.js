import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import AgressorForm from './AgressorForm';

describe('AgressorForm', () => {
  it('deve seguir a estrutura definida ', () => {
    const agressorForm = mount(<AgressorForm />);
    const agressorFormJson = toJson(agressorForm);
    expect(agressorFormJson).toMatchSnapshot();
  });
});

