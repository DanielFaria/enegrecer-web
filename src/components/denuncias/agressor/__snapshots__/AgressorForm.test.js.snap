// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgressorForm deve seguir a estrutura definida  1`] = `
<AgressorForm>
  <div>
    <h3>
      Você se sente confortável em dividir conosco informações sobre o agressor?
    </h3>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc nisl mi, egestas ac ipsum sit amet, blandit posuere metus. Donec imperdiet ut mi sed posuere. Pellentesque et varius sapien.
    </p>
    <div
      className="row"
    >
      <nome
        id="nome-agressor"
        onChange={[Function]}
      >
        <campoTexto
          divClasse="input-field col s6"
          id="nome-agressor"
          inputClasse=""
          label="Nome (máximo de 40 caracteres)"
          maxLen={40}
          onChange={[Function]}
          placeholder=""
          type="text"
        >
          <div
            className="input-field col s6"
          >
            <input
              className=""
              id="nome-agressor"
              maxLength={40}
              onChange={[Function]}
              type="text"
            />
            <label
              className="active"
              htmlFor="nome-agressor"
            >
              Nome (máximo de 40 caracteres)
            </label>
          </div>
        </campoTexto>
      </nome>
      <campoTexto
        divClasse="input-field col s6"
        id="cidade-agressor"
        inputClasse=""
        label="Cidade"
        maxLen={40}
        onChange={[Function]}
        placeholder=""
        type="text"
      >
        <div
          className="input-field col s6"
        >
          <input
            className=""
            id="cidade-agressor"
            maxLength={40}
            onChange={[Function]}
            type="text"
          />
          <label
            className="active"
            htmlFor="cidade-agressor"
          >
            Cidade
          </label>
        </div>
      </campoTexto>
    </div>
    <div
      className="row"
    >
      <genero
        divClasse="input-field col s6"
        id="genero-agressor"
        onChange={[Function]}
      >
        <combobox
          divClasse="input-field col s6"
          id="genero-agressor"
          itens={
            Array [
              "Feminino",
              "Masculino",
            ]
          }
          label="Gênero"
          onChange={[Function]}
          valorPadrao="Selecione seu gênero"
        >
          <div
            className="input-field col s6"
          >
            <select
              id="genero-agressor"
              onChange={[Function]}
            >
              <option
                value=""
              >
                Selecione seu gênero
              </option>
              <option
                key="Feminino"
                value="Feminino"
              >
                 
                Feminino
                 
              </option>
              <option
                key="Masculino"
                value="Masculino"
              >
                 
                Masculino
                 
              </option>
            </select>
            <label
              htmlFor="genero-agressor"
            >
              Gênero
            </label>
          </div>
        </combobox>
      </genero>
      <Data
        divClasse="input-field col s6"
        id="data-nascimento-agressor"
        label="Data de Nascimento"
        onChange={[Function]}
      >
        <campoTexto
          divClasse="input-field col s6"
          id="data-nascimento-agressor"
          inputClasse="datepicker"
          label="Data de Nascimento"
          maxLen={0}
          onChange={[Function]}
          placeholder="Data"
          type="date"
        >
          <div
            className="input-field col s6"
          >
            <input
              className="datepicker"
              id="data-nascimento-agressor"
              maxLength={0}
              onChange={[Function]}
              placeholder="Data"
              type="date"
            />
            <label
              className="active"
              htmlFor="data-nascimento-agressor"
            >
              Data de Nascimento
            </label>
          </div>
        </campoTexto>
      </Data>
    </div>
    <div
      className="row"
    >
      <comboRaca
        id="raca-agressor"
        onChange={[Function]}
        somenteRacasVitima={false}
      >
        <combobox
          divClasse="input-field col s6"
          id="raca-agressor"
          itens={
            Array [
              "Amarela",
              "Branco",
              "Índigena",
              "Parda",
              "Preta",
            ]
          }
          label="Selecione sua raça / cor"
          onChange={[Function]}
          valorPadrao="Selecione sua raça / cor"
        >
          <div
            className="input-field col s6"
          >
            <select
              id="raca-agressor"
              onChange={[Function]}
            >
              <option
                value=""
              >
                Selecione sua raça / cor
              </option>
              <option
                key="Amarela"
                value="Amarela"
              >
                 
                Amarela
                 
              </option>
              <option
                key="Branco"
                value="Branco"
              >
                 
                Branco
                 
              </option>
              <option
                key="Índigena"
                value="Índigena"
              >
                 
                Índigena
                 
              </option>
              <option
                key="Parda"
                value="Parda"
              >
                 
                Parda
                 
              </option>
              <option
                key="Preta"
                value="Preta"
              >
                 
                Preta
                 
              </option>
            </select>
            <label
              htmlFor="raca-agressor"
            >
              Selecione sua raça / cor
            </label>
          </div>
        </combobox>
      </comboRaca>
    </div>
    <div
      className="row"
    >
      <campoTexto
        divClasse="input-field col s12"
        id="descricao-agressor"
        inputClasse=""
        label="Descrição do agressor"
        maxLen={100}
        onChange={[Function]}
        placeholder=""
        type="text"
      >
        <div
          className="input-field col s12"
        >
          <input
            className=""
            id="descricao-agressor"
            maxLength={100}
            onChange={[Function]}
            type="text"
          />
          <label
            className="active"
            htmlFor="descricao-agressor"
          >
            Descrição do agressor
          </label>
        </div>
      </campoTexto>
    </div>
  </div>
</AgressorForm>
`;
